src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,119 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents a List of Tasks, that will be saved to/loaded from
+ * when <code>Duke</code> is run.
+ */
+
+
+public class TaskList {
+    private List<Task> ls;
+
+    public TaskList(List<Task> ls) {
+        this.ls = ls;
+    }
+
+    public TaskList() {
+        this.ls = new ArrayList<>();
+    }
+
+    /**
+     * Lists the string representation of all tasks currently in
+     * the list.
+     */
+    public void listTasks() {
+        for (int i = 0; i < ls.size(); i++) {
+            System.out.println((i + 1) + ". " + ls.get(i));
+        }
+    }
+
+    public String getTasksAsString() {
+        String result = "";
+        for (int i = 0; i < ls.size(); i++) {
+            result = result + (i + 1) + ". " + ls.get(i) + "\n";
+        }
+        return result;
+    }
+
+    /**
+     * Prints tasks in the list that has name containing the specified keyword.
+     * @param keyword
+     */
+    public void findAndPrintTasks(String keyword) {
+        for (int i = 0; i < ls.size(); i++) {
+            if (ls.get(i).getName().contains(keyword)) {
+                System.out.println((i + 1) + ". " + ls.get(i));
+            }
+        }
+    }
+
+    public String findAndGetTasks(String keyword) {
+
+        String result = "";
+
+        for (int i = 0; i < ls.size(); i++) {
+            if (ls.get(i).getName().contains(keyword)) {
+                result +=(i + 1) + ". " + ls.get(i) + "\n";
```

Comment:
This sounds a bit picky, but I suggest adding a space after the equals sign. For such things, I suggest finding the IDE shortcut to autoformat all the files in your project:)